src/Sylius/Component/Core/spec/Model/TaxonSpec.php

```diff
@@ -57,13 +58,39 @@ function it_removes_an_image(ImageInterface $image)
         $this->hasImage($image)->shouldReturn(false);
     }
 
-    function it_returns_an_image_by_code(ImageInterface $image)
+    function it_returns_images_by_type(ImageInterface $image)
     {
-        $image->getCode()->willReturn('thumbnail');
+        $image->getType()->willReturn('thumbnail');
         $image->setOwner($this)->shouldBeCalled();
 
         $this->addImage($image);
 
-        $this->getImageByCode('thumbnail')->shouldReturn($image);
+        $this->getImagesByType('thumbnail')->shouldBeCollection(new ArrayCollection([$image->getWrappedObject()]));
```

Comment:
No need to create custom matchers, `shouldBeLike(new ArrayCollection(...))` should work as well.